src/inter_dc_sub_buf.erl

```diff
@@ -101,15 +105,23 @@ process_queue(State = #state{queue = Queue, last_observed_opid = Last}) ->
 
       %% If the transaction seems to come after an unknown transaction, ask the remote log
         gt ->
-          lager:info("Whoops, lost message. New is ~p, last was ~p. Asking the remote DC ~p",
-		     [TxnLast, Last, State#state.pdcid]),
-          case query(State#state.pdcid, State#state.last_observed_opid + 1, TxnLast) of
-            ok ->
-              State#state{state_name = buffering};
-            _ ->
-              lager:warning("Failed to send log query to DC, will retry on next ping message"),
-              State#state{state_name = normal}
-          end;
+	        case EnableLogging of
+		        true ->
+			        lager:info("Whoops, lost message. New is ~p, last was ~p. Asking the remote DC ~p",
+				        [TxnLast, Last, State#state.pdcid]),
+			        case query(State#state.pdcid, State#state.last_observed_opid + 1, TxnLast) of
+				        ok ->
+					        State#state{state_name = buffering};
+				        _ ->
+					        lager:warning("Failed to send log query to DC, will retry on next ping message"),
+					        State#state{state_name = normal}
+			        end;
+		        false -> %% we deliver the transaction as we can't ask anything to the remote log
```

Comment:
This checks that transactions arrive in total order from a DC to another.
Whenever a transaction arrives an "unknown" one, meanning there is a gap, one will ask the log of the remote DC to resend that transaction. 
When logging is disabled, this can't be done.